plugin/src/main/java/org/opensearch/ml/action/agents/DeleteAgentTransportAction.java

```diff
@@ -39,75 +52,123 @@
+        ClusterService clusterService,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting
     ) {
         super(MLAgentDeleteAction.NAME, transportService, actionFilters, MLAgentDeleteRequest::new);
         this.client = client;
+        this.sdkClient = sdkClient;
         this.xContentRegistry = xContentRegistry;
         this.clusterService = clusterService;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<DeleteResponse> actionListener) {
         MLAgentDeleteRequest mlAgentDeleteRequest = MLAgentDeleteRequest.fromActionRequest(request);
         String agentId = mlAgentDeleteRequest.getAgentId();
+        String tenantId = mlAgentDeleteRequest.getTenantId();
         GetRequest getRequest = new GetRequest(ML_AGENT_INDEX).id(agentId);
         boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = new GetDataObjectRequest.Builder()
+            .index(ML_AGENT_INDEX)
+            .id(agentId)
+            .tenantId(tenantId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
+
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            client.get(getRequest, ActionListener.wrap(getResponse -> {
-                if (getResponse.isExists()) {
-                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, getResponse.getSourceAsBytesRef())) {
-                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                        MLAgent mlAgent = MLAgent.parse(parser);
-                        if (mlAgent.getIsHidden() && !isSuperAdmin) {
-                            actionListener
-                                .onFailure(
-                                    new OpenSearchStatusException(
-                                        "User doesn't have privilege to perform this operation on this hidden agent",
-                                        RestStatus.FORBIDDEN
-                                    )
-                                );
+
+            sdkClient
+                .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                .whenComplete((r, throwable) -> {
+                    context.restore();
+                    log.debug("Completed Get Agent Request, Agent id:{}", agentId);
+                    if (throwable != null) {
+                        Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
+                        if (cause instanceof IndexNotFoundException) {
+                            log.error("Failed to get Agent index", cause);
```

Comment:
This error log seems not helpful, because this is expected and not server side error